lib/stack_master/prompter.rb

```diff
@@ -1,11 +1,17 @@
 module StackMaster
   module Prompter
     def ask?(question)
-      StackMaster.stdout.print question
-      answer = if ENV['STUB_AWS']
-        ENV['ANSWER']
+      answer = if StackMaster.interactive?
+        StackMaster.stdout.print question
+        if StackMaster.stdin.tty? && StackMaster.stdout.tty?
+          StackMaster.stdin.getch.chomp
+        else
+          StackMaster.stdout.puts
+          StackMaster.stdout.puts "STDOUT or STDIN was not a TTY. Defaulting to no. To force yes use -f"
+          'n'
+        end
       else
-        STDIN.getch.chomp
+        ENV.fetch('ANSWER') { 'y' }
```

Comment:
Why not use another command line option to answer questions, rather than an environment variable?